File: solo-leveling-player-service/src/main/java/com/sleepkqq/sololeveling/player/service/exception/ModelNotFoundException.java
package com.sleepkqq.sololeveling.player.service.exception;

import com.sleepkqq.sololeveling.player.service.model.Model;

public class ModelNotFoundException extends RuntimeException {

  public <ID> ModelNotFoundException(Class<? extends Model> clazz, ID id) {
    super(clazz.getSimpleName() + " not found id=" + id);
  }
}

File: solo-leveling-player-service/src/main/kotlin/com/sleepkqq/sololeveling/player/service/exception/ModelNotFoundException.kt
package com.sleepkqq.sololeveling.player.service.exception

import com.sleepkqq.sololeveling.player.model.entity.Model
import kotlin.reflect.KClass

class ModelNotFoundException(klass: KClass<out Model>, id: Any) :
	RuntimeException("${klass.simpleName} not found id=$id")
